src/OpenLoco/src/GameState.h

```diff
@@ -138,7 +138,7 @@ namespace OpenLoco
         uint8_t orders[Limits::kMaxOrders];                                      // 0x461E44 (0x00987C5C)
     };
 #pragma pack(pop)
-    static_assert(sizeof(GameState) == 0x4A0644);
+    static_assert(sizeof(GameState) == 0x4A0642);
```

Comment:
The size of the GameState can't change due to these changes.

Reply:
The original GameState used a uint32_t for the flags, but the Scenario::Flags used uint16_t. Which type should I use?

Reply:
The memory offsets can't change due to interop with Locomotion code. Please use a uint32_t instead.

Reply:
The original S5.h file also uses a uint16_t for scenario flags. When I change to uint32_t, it causes the assertions to fail. Should I change those?

Reply:
You're right. Since the upper parts of the `uint32_t` weren't used anyway, I think the best course of action would be to instead add some padding to the GameState, e.g.:
```diff
-       uint32_t flags;                                                          // 0x000010 (0x00525E28)
+       Scenario::Flags flags;                                                   // 0x000010 (0x00525E28)
+       uint16_t pad_12                                                          // 0x000012 (0x00525E2A)
        uint32_t currentDay;                                                     // 0x000014 (0x00525E2C)

Reply:
I think the actual issue is that these (the gamestate::flags) aren't scenario flags. This is one of the reasons for making these strong types it identifies these issues and these look to not be related. I've checked the assembly and the gamestate::flags are never copied into the S5::Options::scenarioFlags or vice versa. They also appear to be used differently the gamestate::flags should really have the following:
```
        constexpr uint32_t tileManagerLoaded = (1 << 0);
        constexpr uint32_t unk2 = (1 << 1);
        constexpr uint32_t preferredOwnerName = (1 << 2);
```
and S5::Options::scenarioFlags should have
```
        constexpr uint16_t landscapeGenerationDone = (1 << 0);
        constexpr uint16_t hillsEdgeOfMap = (1 << 1);
```

Reply:
See also https://github.com/OpenLoco/OpenLoco/issues/1517 https://github.com/OpenLoco/OpenLoco/pull/1537